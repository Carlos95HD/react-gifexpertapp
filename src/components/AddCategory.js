import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategories} ) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    // console.log( e.target.value )
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    // console.log('Submit')
    if ( inputValue.trim().length > 2 ) {
      setCategories( catg => [inputValue,...catg] )
      setInputValue('')
    }
  }


  return (
    <form onSubmit={ handleSubmit }>
      <input 
      className='mb-4 mt-2 shadow-lg p-2 font-mono rounded focus:outline-none'
      placeholder='Search'
      type='text'
      value= { inputValue }
      onChange={ handleInputChange }
      />

    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}