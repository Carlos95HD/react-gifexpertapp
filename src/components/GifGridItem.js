import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( { title, url} ) => {

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={ title } className='text-center'/>
      <p className='font-sans text-gray-800'>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}