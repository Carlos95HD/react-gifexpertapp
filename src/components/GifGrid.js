import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import {GifGridItem }from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data:images, loading} = useFetchGifs(category);

  // getGifs();

  return (
    <>
    <h3 className='animate__animated animate__fadeIn font-mono mb-5 mt-5 text-base text-2xl text-gray-800'>{category}</h3>

    {loading && <p className='font-mono animate__animated animate__flash text-gray-800'>Loading</p>}

    <div className='card-grid'>

      {
        images.map( img  => (
          <GifGridItem
          key={img.id}
          {...img}
          />
        ))
      }

    </div>
    </>

  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}