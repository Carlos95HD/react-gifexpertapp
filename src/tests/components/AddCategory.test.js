import React from 'react'
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('pruebas en <AddCategory />', () => {

  const setCategories = jest.fn();
  let wrapper = shallow( <AddCategory setCategories = { setCategories }/> )

  beforeEach( () => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategories = { setCategories }/> )
  });

  test('Se debe mostrar correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  })

  // test('debe cambiar la caja de texto', () => {
  //   const input = wrapper.find('input');
  //   const value = 'Hola Mundo'
  //   input.simulate( 'change', { target: {value} } )

  //   expect( wrapper.find('p').text().trim() ).toBe( value )
  // })

  test('No debe postear la información del submit', () => {

    wrapper.find('form').simulate( 'submit', { preventDefault(){} } );

    expect( setCategories ).not.toHaveBeenCalled();
  })

  test('se debe llamar el setCategories y limpiar la caja de texto', () => {

    const value = 'Hola'
    // 1 simultar el inputChange
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });

    // 2 simular el submit
    wrapper.find('form').simulate( 'submit', { preventDefault(){} } );

    // 3 setCategories se debe de haber llamado
    expect(setCategories).toHaveBeenCalled();

    //Funcion para saber las cantidad de veces ha sido llamada
    expect(setCategories).toHaveBeenCalledTimes(1);
    //Espera una función a travez
    expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );

    // 4 el valor del input debe de estar ''
    expect( wrapper.find('input').prop('value')).toBe('');
  });


});