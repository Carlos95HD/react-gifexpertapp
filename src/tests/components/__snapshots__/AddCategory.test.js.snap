// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AddCategory /> Se debe mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="mb-4 mt-2 shadow-lg p-2 font-mono rounded focus:outline-none"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
